Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -6,6 +6,8 @@ def check_shadowed_headers
       formula.name.start_with?(formula_name)
     end
 
+    return if formula.name =~ /^php[57]\d$/
```

Comment:
I'm thinking it will change if we move into the new versioned model with php@... as well. I just wanted it to be as specific as it could be but I can change it and its likely that it won't cause any clashes anyhow. I'll change it.